src/ir/branch-utils.h

```diff
@@ -75,6 +75,13 @@ void operateOnScopeNameUsesAndSentTypes(Expression* expr, T func) {
       func(name, sw->value ? sw->value->type : Type::none);
     } else if (auto* br = expr->dynCast<BrOn>()) {
       func(name, br->getSentType());
+    } else if (auto* tt = expr->dynCast<TryTable>()) {
+      for (Index i = 0; i < tt->catchTags.size(); i++) {
+        auto dest = tt->catchDests[i];
+        if (dest == name) {
```

Comment:
Why is this comparing to `name`? The name should only be provided to `func` (which can modify it) IIUC.

Reply:
Sorry I think I was confused. Fixed it.

Reply:
No I think the original code was correct... `name` given here is a block label, and we should run `func` only when the `name` is the label we are targeting. For example:
```wast
(block $l-catch-ref (result i32 exnref)
  ...
  (block $l-catch (result i32)
    ...
    (try_table (catch $e-i32 $l-catch) (catch_ref $e-i32 $l-catch-ref)
      ...
    )
  )
)
```
In this case, when analyzing `TryTable`, we shouldn't send both `i32` and `(i32 exnref)` to a given `name`, which will mean the given `name` has two different types. We send the type only when the given `name` matches its `sentType`'s label.

Reply:
I see, thanks. So this is called multiple times for a single `TryTable`, once per name, and we need to handle the right name each time.

Maybe worth handling `TryTable` outside of the `operateOnScopeNameUses` to avoid the potentially quadratic work? But if we think that will be incredibly rare maybe just a comment is enough.

Reply:
Even though theoretically there can be any number of `catch`es, in practice there is no reason to use more than two (one for one of `catch`/`catch_ref` and the other for one of `catch_all`/`catch_all_ref`), unless you are trying to catch multiple language tags. And in LLVM we don't even generate two `catch`es: We only generate one `catch` or `catch_all` for a single `try` due to the structure of Windows EH IR, and I think this will be also the case for the new spec. Even though we assume a non-LLVM multi-language handling case I doubt this will go more than 3.